### PR TITLE
Update the validate deliveries table 

### PR DESCRIPTION
**Pull request by tarrsalah**
This PR was required by evercam/evercam-server2#1500 .

### CODE REVIEW

File: lib/evercam_models/analytics/deliveries.ex [hunk 1]
```diff
@@ -202,4 +209,41 @@ defmodule Analytics.Deliveries do
...
+        :arrived_time,
+        :left_time,
+        :truck_type
+      ])
+      |> put_change(:detection_type, @detection_type_auto)
+
+    insert_changeset = merge(old_event_changeset, new_event_changeset)
+
+    Multi.new()
+    |> Multi.update(:update, update_changeset)
```
Comment (ijunaid8989) [inline lib/evercam_models/analytics/deliveries.ex:245]: why this instead of this? https://hexdocs.pm/ecto/Ecto.Multi.html#insert_or_update/4

Comment (tarrsalah) [reply lib/evercam_models/analytics/deliveries.ex:245]: What I need is a different semantic:
1- Update a row (`detection_type` field from `test` to `corrected`).
2- Then insert a **new** row (inherited  from the update row in step 1) with a new params ( `:arrived_time` ,
        `:left_time` and  `:truck_type`) and `detection_type` `auto`.

So it is not an insert **or** update operation, but update **and** insert within a transaction.

Comment (ijunaid8989) [reply lib/evercam_models/analytics/deliveries.ex:245]: Awesome. Nice one.

**Comment by ijunaid8989** [comment]
its perfect. and that Multi one is a nice one.

**Comment by tarrsalah** [comment]
Closed for #48.

**Pull request closed without merging**